### PR TITLE
Allow repair of Terrainified station due to shuttles and mining magnet

### PR DESCRIPTION
**Pull request by Azrun**
[bug][feature]
## About the PR 
Store aspects of -ify verbs to allow for various things to "repair" the station z-level by regenerating turfs.

https://streamable.com/5855kj
https://streamable.com/u9m3o4

## Why's this needed? 
image

Make it so mining magnet still work post -ifyication

### CODE REVIEW

**Comment by Azrun** [comment]
Oh, forgot mining shuttle 🤦

**Comment by Azrun** [comment]
This should resolve issues with use of Terrain-ify based verbs but also is a ticking time bomb of merge conflicts for Artemis, Theta Generator, and Mars-ify.

Would love to get the station in a more fully functioning state when these admemes are utilized.

**Review by Sovexe — APPROVED** [review]

File: code/datums/mapgen/_MapGenerator.dm [hunk 1]
```diff
@@ -9,9 +9,14 @@ proc/initialize_biomes()
 
 ///This type is responsible for any map generation behavior that is done in areas, override this to allow for area-specific map generation. This generation is ran by areas on world/proc/init().
 /datum/map_generator
+	var/seeds = list()
 
 ///This proc will be ran by areas on world/proc/init(), and provides the areas turfs as argument to allow for generation.
-/datum/map_generator/proc/generate_terrain(var/list/turfs)
+/datum/map_generator/proc/generate_terrain(var/list/turfs, var/reuse_seed)
+	if(!reuse_seed || !length(seeds))
```
[the remark this answers is not in the document]
Comment (Azrun) [reply code/datums/mapgen/_MapGenerator.dm:16]: Correct!

**Review by Tarmunora — APPROVED** [review]

File: code/datums/mapgen/_MapGenerator.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Azrun) [reply code/datums/mapgen/_MapGenerator.dm:16]: Added handling for null for good measure.

File: code/datums/mapgen/_MapGenerator.dm [hunk 2]
```diff
@@ -9,9 +9,16 @@ proc/initialize_biomes()
 
 ///This type is responsible for any map generation behavior that is done in areas, override this to allow for area-specific map generation. This generation is ran by areas on world/proc/init().
 /datum/map_generator
+	var/seeds = list()
```
Comment (Tarmunora) [inline code/datums/mapgen/_MapGenerator.dm:12]:
```suggestion
	var/list/seeds
```

**Pull request merged**